reader_concurrency_semaphore.cc

```diff
@@ -749,6 +749,18 @@ void reader_concurrency_semaphore::clear_inactive_reads() {
     }
 }
 
+future<> reader_concurrency_semaphore::evict_inactive_reads_for_table(table_id id) noexcept {
+    auto it = _inactive_reads.begin();
+    while (it != _inactive_reads.end()) {
+        auto& ir = *it;
+        ++it;
+        if (ir.reader.schema()->id() == id) {
+            auto reader = detach_inactive_reader(ir, evict_reason::manual);
+            co_await reader.close();
```

Comment:
While we're waiting for this close(), what's keeping `it` alive? It could have self-evicted due to the timer, or memory pressure, no?

Reply:
(I think `reader` itself is protected by `detach_inactive_reader()`, but the next one isn't).

Reply:
Ah, you mean `it` could be invalidated while we wait for the reader to close. Yes, nothing prevents that currently, I have to make sure we avoid this.